### ISSUE TITLE
Dataset Size in Params.JSON for Training

### ISSUE THREAD

**Issue by cloudlakecho** [micmelesse/thesis#4]
I try to reproduce the result from [Michael's paper](https://github.com/micmelesse/3D-reconstruction-with-Neural-Networks/blob/master/3D_reconstruction_with_neural_networks.pdf).

On his paper page 27, it looks like the training conducted with these parameter;

Batch Size: 16
Epoch: 40
(Max) Iteration: 30,000

With this parameter, the dataset size should be around 12,000.

Anybody or @micmelesse train the RNN with these parameters?

**Comment by micmelesse** [comment]
I think the training session on page 27 goes on for 80,000 steps. As to the DATA_SIZE parameter I usually just set it to 0. This will make it use all of the available data (around 50,000 samples) to construct the training, val and test sets. Let me know if this helps.

**Comment by cloudlakecho** [comment]
@micmelesse, thanks for the details of training parameters.

It looks like your training parameters were;

Batch Size:16
Epoch: 26
(Max) Iteration: 80,000
DATA_SIZE: 0 (50,000)

According to your paper page 27, after 30,000 steps there is divergence. Do you remember if there had a continuous improvement in your Y_p (Y prediction) after 30,000 iteration?  

Another question is about number of epoch, [your sample video on Youtube](https://www.youtube.com/watch?v=iI6ZMST8Ri0) looks like from 40 epoch. Did you choose 40 epoch and reduced batch size to 25? 

Batch Size:25
Epoch: 40
(Max) Iteration: 80,000
DATA_SIZE: 0 (50,000)

**Comment by micmelesse** [comment]
The session on page 27 went on for 40 Epochs. The batch size was 16 or smaller. Bigger batches could not fit on the GPU that I was using. As you mentioned there is a divergence between the training and validation error meaning that predictions from the training set continuously get better as the model overfits but predictions from the validation set stopped improving after 30000th iteration.

**Comment by cloudlakecho** [comment]
@micmelesse, thanks again for the details of training parameters.

It looks like your training parameters were;

Batch Size:16
Epoch: 40
(Max) Iteration: 125,000
DATA_SIZE: 0 (50,000)

I still have questions because I want to train your neural network correctly. Was smaller data size used to reduce iteration? I know it probably hard to remember details, but I will appreciate training setting information.

**Comment by micmelesse** [comment]
Can you clarify your question? I am not sure I quite understand.

**Comment by cloudlakecho** [comment]
Sorry for confusion.

If you keep epoch, 40 and batch size 16, the (maximum) iteration should go up to 125,000 
(from the equation, (data size / batch size) x epoch = iteration). Was that your training environment for page 27 in your paper?

**Comment by micmelesse** [comment]
That seems about right. I used a batch size of 16 because it was what would fit in the GPU that I was using.  Each sample is a sequence of images that could be up to 24 so I had to keep the batch size low. If you have a bigger GPU then I don't see why you cannot give it a try.

**Comment by cloudlakecho** [comment]
I received detailed answers.
